examples/gin/main.go

```diff
@@ -0,0 +1,136 @@
+)
+
+var ProjectId int64 = 105138                               // Replace it with your project Id
+var ProjectKey string = "fd04e13d806a90f96614ad8e529b2822" // Replace it with your project key
+
+// Define models
+
+type Date struct {
+	Date int64 // the server date
+}
+
+type WeatherInfo struct {
+	Lat            float32       `json:"lat,omitempty"`
+	Lon            float32       `json:"lon,omitempty"`
+	Timezone       string        `json:"timezone,omitempty"`
+	TimezoneOffset float32       `json:"timezone_offset,omitempty"`
+	Current        interface{}   `json:"current,omitempty"`
+	Minutely       []interface{} `json:"minutely,omitempty"`
+	Hourly         []interface{} `json:"hourly,omitempty"`
+	Daily          []interface{} `json:"daily,omitempty"`
+}
+
+var availableLocations = []string{
+	"austin",
+	"pune",
+	"santabarbara",
+}
+
+var notifier = gobrake.NewNotifierWithOptions(&gobrake.NotifierOptions{
+	ProjectId:   ProjectId,    // use it as an environment variable
+	ProjectKey:  ProjectKey,   // use it as an environment variable
+	Environment: "production", // use it as an environment variable
+})
+
+func main() {
+
+	api := gin.Default()
+
+	// Initialise middlewares
+	api.Use(ginbrake.New(api, notifier))
+	api.Use(TokenAuthMiddleware())
+
+	// Initialise routes
+	api.GET("/date", getDate)
+	api.GET("/locations", getLocations)
+	api.GET("/weather/:location", getWeather)
+
+	// Initialise application
+	api.Run(":3000")
+}
+
+// Define handlers for routes
+
+func getDate(c *gin.Context) {
+	fmt.Println(c.Request.Header.Get("api-key"))
+	now := time.Now()
+	secs := now.Unix()
+	c.JSON(http.StatusOK, gin.H{"date": secs})
+}
+
+func getLocations(c *gin.Context) {
+	c.JSON(http.StatusOK, availableLocations)
+}
+
+func getWeather(c *gin.Context) {
+	var weatherInfo WeatherInfo
+	location := strings.TrimSpace(c.Param("location")) // It lets you allow whitespaces but case-sensitive
+	if !contains(availableLocations, location) {
+		c.JSON(http.StatusNotFound, weatherInfo)
+		return
+	}
+	weatherInfoFile := location + ".json"
+	weatherInfoFilePath := filepath.Join("asset", weatherInfoFile)
+	// Open our jsonFile
+	jsonFile, err := os.Open(weatherInfoFilePath)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		notifier.Notify(err, nil)
+		c.JSON(http.StatusNoContent, weatherInfo)
+		return
+	}
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	json.Unmarshal(byteValue, &weatherInfo)
+	// defer the closing of our jsonFile so that we can parse it later on
+	defer jsonFile.Close()
+	c.JSON(http.StatusOK, weatherInfo)
+}
+
+// Function to check whether it is a valid location
+func contains(a []string, x string) bool {
+	for _, n := range a {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+func respondWithError(c *gin.Context, code int, message interface{}) {
+	c.AbortWithStatusJSON(code, gin.H{"error": message})
+}
+
+func TokenAuthMiddleware() gin.HandlerFunc {
+	requiredToken := "d4b371692d361869183d92d84caa5edb8835cf7d"
```

Comment:
What is the significance of this token for this example app?

Reply:
It is built according to the swagger api specification given by @scottsbaldwin.

Reply:
@scottsbaldwin would you mind taking a look? The token stuff might be omitted without hurting the example app experience.

Reply:
I think we should keep it.